src/Models/ElementalArea.php

```diff
@@ -33,6 +37,10 @@ class ElementalArea extends DataObject
         'Elements' => BaseElement::class,
     ];
 
+    private static $has_one = [
+        'TopPage' => Page::class,
```

Comment:
The fact that we need to add the DB stuff on the implementing models now, makes me think we shouldn't be using a trait here and instead we should just be using extensions.  Extensions are basically the Silverstripe+ version of traits.

Reply:
I agree, I'd prefer to see this as a data extension.